File: Web/frontend/src/services/settings.service.ts
import { useQuery } from "@tanstack/react-query";
import { api } from "@/services/api";
import type { SettingsResponse } from "@/types/api";

/** GET /api/settings — courier connections, rate-card meta, read-only system preferences. */
export function useSettings() {
  return useQuery({
    queryKey: ["settings"],
    queryFn: async () => (await api.get<SettingsResponse>("/settings")).data,
  });
}

File: Web/backend/app/services/settings_service.py
"""Configuration/settings service — READ-ONLY.

Serves the courier roster, rate-card meta, and read-only system preferences.
Nothing is editable: the Company block was fabricated data and the notification
toggles saved nowhere (no email/alert system reads them), so both were removed
along with the PATCH endpoint.
"""

from app.schemas.settings import SettingsResponse
from app.utils.mock import load_mock

_settings: dict = load_mock("settings.json")


def get_settings() -> SettingsResponse:
    return SettingsResponse(**_settings)
